modules/mod_login/Helper/LoginHelper.php

```diff
@@ -63,4 +63,35 @@ public static function getType()
 
 		return (!$user->get('guest')) ? 'logout' : 'login';
 	}
+
+	/**
+	 * Retrieve the URL for the registration page
+	 *
+	 * @return  string
+	 */
+	public static function getRegistrationUrl($params)
+	{
+		$regLinkMenuId = $params->get('customRegLinkMenu', 0);
+
+		// If there is no custom menu item set for registration => use the default link
+		if ($regLinkMenuId == '' || $regLinkMenuId == '0')
+		{
+			$regLink = 'index.php?option=com_users&view=registration';
+		}
+		else
+		{
+			$regLink = 'index.php?Itemid=' . $regLinkMenuId;
+			$item = Factory::getApplication()->getMenu()->getItem($regLinkMenuId);
+
+			if ($item)
+			{
+				if ($item->language !== '*' && Multilanguage::isEnabled())
+				{
+					$regLink .= '&lang=' . $item->language;
+				}
+			}
+		}
+
+		return base64_encode($regLink);
```

Comment:
```suggestion
		return $regLink;
```
Don't need to encode here.